coriolis/tests/test_wsman.py

```diff
@@ -0,0 +1,157 @@
+            self.cmd, self.args)
+        self.assertEqual(std_out, "std_out")
+        self.assertEqual(std_err, "std_err")
+        self.assertEqual(exit_code, 0)
+
+        self.conn._protocol.open_shell.assert_called_once_with(
+            codepage=wsman.CODEPAGE_UTF8)
+        shell_id = self.conn._protocol.open_shell.return_value
+        self.conn._protocol.run_command.assert_called_once_with(
+            shell_id, self.cmd, self.args)
+        command_id = self.conn._protocol.run_command.return_value
+        self.conn._protocol.get_command_output.assert_called_once_with(
+            shell_id, command_id)
+
+    def test__exec_command_exception(self):
+        self.conn._protocol.get_command_output.side_effect = requests.\
+            exceptions.ReadTimeout
+        self.assertRaises(exception.OSMorphingWinRMOperationTimeout,
+                          self.conn._exec_command, self.cmd, self.args)
+
+    def test_exec_command(self):
+        self.conn._protocol.get_command_output.return_value = (
+            "std_out", "std_err", 0)
+        std_out = self.conn.exec_command(self.cmd, self.args)
+        self.assertEqual(std_out, "std_out")
+
+    def test_exec_command_exception(self):
+        self.conn._protocol.get_command_output.return_value = (
+            "std_out", "std_err", 1)
+        self.assertRaises(exception.CoriolisException, self.conn.exec_command,
+                          self.cmd, self.args)
+
+    def test_exec_ps_command(self):
+        self.conn.exec_command = mock.Mock()
+        self.conn.exec_command.return_value = "std_out\n\n"
+        result = self.conn.exec_ps_command(self.cmd)
+        self.conn.exec_command.assert_called_once_with(
+            "powershell.exe", ["-EncodedCommand", mock.ANY], timeout=None)
+        self.assertEqual(result, "std_out")
+
+    def test_test_path(self):
+        self.conn.exec_ps_command = mock.Mock()
+        self.conn.exec_ps_command.return_value = "True"
+        result = self.conn.test_path("test_path")
+        self.conn.exec_ps_command.assert_called_once_with(
+            "Test-Path -Path \"test_path\"")
+        self.assertTrue(result)
+
+    def test_download_file(self):
+        self.conn.exec_ps_command = mock.Mock()
+        self.conn.download_file(self.url, self.remote_path)
+        self.conn.exec_ps_command.assert_called_once()
+
+    def test_download_file_exception(self):
+        self.conn.exec_ps_command = mock.Mock()
+        self.conn.exec_ps_command.side_effect = exception.CoriolisException
+        self.assertRaises(exception.CoriolisException, self.conn.download_file,
+                          self.url, self.remote_path)
+        self.conn.exec_ps_command.assert_called_once()
+
+    def test_write_file(self):
+        self.conn.exec_ps_command = mock.Mock()
+        self.conn.write_file(self.remote_path, b'file content')
+        self.conn.exec_ps_command.assert_called_once_with(
+            mock.ANY, ignore_stdout=True)
```

Comment:
Same as above, add encoded content instead of `mock.ANY`:

`'ZmlsZSBjb250ZW50'`